File: src/ELibrary.Backend/LibraryShopEntities/Domain/Entities/Library/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryShopEntities.Domain.Entities.Library
{
    public enum CoverType
    {
        Any = 0, Hard, Soft
    }
    public class Book : BaseLibraryEntity
    {
        [Required]
        public DateTime PublicationDate { get; set; }
        [Required]
        [Column(TypeName = "decimal(18, 2)")]
        public decimal Price { get; set; }
        [Required]
        public CoverType CoverType { get; set; }
        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "Page amount must be greater than 0")]
        public int PageAmount { get; set; }
        [Range(0, int.MaxValue, ErrorMessage = "Stock amount must be greater than or equal to 0")]
        public int StockAmount { get; set; }
        [Required]
        [MaxLength(1024)]
        public string CoverImgUrl { get; set; } = default!;
        public BookPopularity? BookPopularity { get; set; } = default!;
        [Required]
        public int AuthorId { get; set; }
        [ForeignKey("AuthorId")]
        public Author Author { get; set; } = default!;
        [Required]
        public int GenreId { get; set; }
        [ForeignKey("GenreId")]
        public Genre Genre { get; set; } = default!;
        [Required]
        public int PublisherId { get; set; }
        [ForeignKey("PublisherId")]
        public Publisher Publisher { get; set; } = default!;

        public override void Copy(BaseLibraryEntity other)
        {
            if (other is Book otherBook)
            {
                Name = otherBook.Name;
                PublicationDate = otherBook.PublicationDate;
                Price = otherBook.Price;
                CoverType = otherBook.CoverType;
                PageAmount = otherBook.PageAmount;
                CoverImgUrl = otherBook.CoverImgUrl;
                AuthorId = otherBook.AuthorId;
                GenreId = otherBook.GenreId;
                PublisherId = otherBook.PublisherId;
            }
        }
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/models/library/book.ts
import { Author, Genre, getDefaultAuthor, getDefaultGenre, getDefaultPublisher, Publisher } from "../../..";

export enum CoverType {
    Any = 0, Hard = 1, Soft = 2
}
export interface Book {
    id: number;
    name: string;
    publicationDate: Date;
    price: number;
    coverType: CoverType;
    pageAmount: number;
    stockAmount: number;
    coverImgUrl: string,
    author: Author;
    genre: Genre;
    publisher: Publisher;
}
export function getDefaultBook(): Book {
    return {
        id: 0,
        name: "",
        publicationDate: new Date(),
        price: 0,
        coverType: CoverType.Hard,
        pageAmount: 0,
        stockAmount: 0,
        coverImgUrl: "",
        author: getDefaultAuthor(),
        genre: getDefaultGenre(),
        publisher: getDefaultPublisher()
    }
}
export function getStringCoverType(coverType: CoverType): string {
    switch (coverType) {
        case CoverType.Hard:
            return 'Hard';
        case CoverType.Soft:
            return 'Soft';
        case CoverType.Any:
        default:
            return 'Any';
    }
}